src/asphalt/core/cli.py

```diff
@@ -67,6 +80,37 @@ def run(configfile, unsafe: bool, loop: Optional[str], service: Optional[str]) -
         ), "the document root element must be a dictionary"
         config = merge_config(config, config_data)
 
+    # Read the configuration from the CLI
+    for cli_conf in set_:
+        if "=" not in cli_conf:
+            click.echo(f"Configuration must be set with '=', got: {cli_conf}")
+            raise click.Abort()
+
+        key, value = cli_conf.split("=", 1)
```

Comment:
I think the value should be interpreted as YAML. Otherwise there is no way to set non-string values.

Reply:
That's what I thought, but it actually works fine (see [my comment](https://github.com/asphalt-framework/asphalt/pull/50#discussion_r1032017224)).

Reply:
Where is Asphalt supposedly casting the value? You don't even have any tests confirming that.

Reply:
I was trying it on an example and saw that configuring an `int` component argument worked fine. I don't know where casting is done, I'll try to add a test.

Reply:
Fixed in https://github.com/asphalt-framework/asphalt/pull/50/commits/cea126c29631edde6bc497ce184f036922c9330a.

Reply:
I don't quite like this approach. I'll make my own suggestion some time this weekend.

Reply:
I pushed a changeset. Let me know if you are opposed to the changes I made.